File: lab/python/tecnicas/inverter_numeros.py
def inverter_numeros1(numero):
    negativo = numero < 0
    inverso = str(numero)[::-1].replace("-", "")
    return -int(inverso) if negativo else int(inverso)


def inverter_numeros2(numero):
    negativo = numero < 0  # Verifica se é negativo
    numero = abs(numero)   # Trata negativos

    inverso = 0

    while numero > 0:
        resto = numero % 10
        inverso = (inverso * 10) + resto
        numero //= 10

    return -inverso if negativo else inverso


print(inverter_numeros1(1234))
print(inverter_numeros2(1234))
print()
print(inverter_numeros1(-1234))
print(inverter_numeros2(-1234))
print()
print(inverter_numeros1(0))
print(inverter_numeros2(0))
print()
print(inverter_numeros1(1000))
print(inverter_numeros2(1000))

File: lab/javascript/tecnicas/inverter_numeros.js
function inverter_numeros1(numero) {
  const negativo = numero < 0;
  const inverso = Math.abs(numero).toString().split("").reverse().join("");
  return negativo ? -parseInt(inverso) : parseInt(inverso);
}

function inverter_numeros2(numero) {
  const negativo = numero < 0;
  numero = Math.abs(numero);
  let inverso = 0;

  while (numero > 0) {
    const resto = numero % 10;
    inverso = inverso * 10 + resto;
    numero = Math.floor(numero / 10);
  }

  return negativo ? -inverso : inverso;
}

// -------------------- Teste --------------------
console.log(inverter_numeros1(1234));
console.log(inverter_numeros2(1234));
console.log();
console.log(inverter_numeros1(-1234));
console.log(inverter_numeros2(-1234));
console.log();
console.log(inverter_numeros1(0));
console.log(inverter_numeros2(0));
console.log();
console.log(inverter_numeros1(1000));
console.log(inverter_numeros2(1000));
